### ISSUE TITLE
rustup allows `rustup toolchain link 'toolchain with spaces'`, then panics

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rustup#3168]
### Problem

```
; rustup toolchain link 'toolchain with spaces' ~/src/rust/build/x86_64-apple-darwin/stage0
; rustup toolchain list
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected character in pre-release identifier")', src/utils/utils.rs:547:83
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
; rustc '+toolchain with spaces' --version
rustc 1.67.0-beta.3 (b364405cc 2022-12-27)
```

### Steps

1. rustup toolchain link 'toolchain with spaces' ~/src/rust/build/x86_64-apple-darwin/stage0
2. rustup toolchain list

### Possible Solution(s)

Either rustup should not allow creating a toolchain that has spaces in the name, or rustup shouldn't panic on encountering the name in `toolchain list`.

### Notes

cc https://github.com/rust-lang/rustup/issues/3130

### Rustup version

```console
rustup 1.25.1 (bb60b1e89 2022-07-12)
```

### Installed toolchains

```console
; rustup show
Default host: x86_64-apple-darwin
rustup home:  /Users/jyn/.local/lib/rustup

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected character in pre-release identifier")', src/utils/utils.rs:547:83
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
; ls /Users/jyn/.local/lib/rustup/toolchains
1.65-x86_64-apple-darwin    stable-x86_64-apple-darwin
nightly-x86_64-apple-darwin toolchain with spaces
```

**Comment by rbtcollins** [comment]
I see no reason not to support paths with spaces. Looks fairly straight forward to fix.

**Comment by ashutoshvarma** [comment]
@jyn514 would it be alright if I include the fix for this in #3174 also? Both are very small changes.

**Comment by jyn514** [comment]
@ashutoshvarma I don't maintain rustup so you should ask @rbtcollins, but that seems fine to me personally :)

**Comment by rbtcollins** [comment]
Thinking on this, I think I mis-parsed initially.

The /name/ should reject spaces, categorically. We have a function to test sensible names but its clearly not paranoid enough. Patches would be appreciated!

**Comment by rami3l** [comment]
Looks like this is a duplicate of #3517, which has been fixed by #3518. As such, I'm closing this issue now. Please feel free to reopen if this is not the case.
